packages/dapp/src/components/HighlightModal/Footer.js

```diff
@@ -13,18 +15,43 @@ const greenButtonDisabledStyle = `w-32 px-4 py-2 bg-gray-600 shadow shadow-green
 
 const Footer = () => {
   const { state, dispatch } = useContext(HighlightContext);
-  const [{ data, error }, connect] = useConnect()
+  const {
+    activeConnector,
+    connect,
+    connectors,
+    error,
+  } = useConnect();
+  const { data: signer } = useSigner();
+  const signatureNFTContract = useContract({
+    addressOrName: addresses(1337).signatureNFT, // @todo - need a better way to manage this, the chainid from provider will be a promise, and probably can't be used here
```

Comment:
since fetching chainid from anywhere would require us to `await`, this seemed easier, open to suggestions here

Reply:
You don't need to do any of this though?

All of it is already instantiated in `utils/contracts.js` afaik. Just 

```
import { mintSelected } from '../../utils/contracts`;
...
const provider = useProvider()
...
await mintSelected(metadataUrl, provider, signer)
```

as it was - all you need to figure out with the upgrade is what to pass instead of the provider.

Reply:
Oh but I believe (I could be wrong here) that sending a variable from the `useProvider` hook would be antipattern, since that's a state variable, and might be undefined as soon as the page loads, we won't be able to control the flow of execution on our custom `mintSelected` function (we can't force it to **wait** for an object to be defined to execute)

Reply:
ok switched to util functions in https://github.com/kernel-community/signature-economies/pull/9 👍🏼